src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -1275,7 +1274,9 @@ private function copyFromHtml5ToDom(\Dom\Node $source, \DOMDocument $target): vo
                 try {
                     $element = $target->createElement($source->tagName);
                 } catch (\DOMException) {
-                    continue;
+                    $dom = new \DOMDocument('1.0', $target->encoding);
+                    $dom->loadHTML('<'.$source->tagName.'>', \LIBXML_HTML_NOIMPLIED | \LIBXML_HTML_NODEFDTD);
```

Comment:
I'm not sure I fully understand what this code is supposed to do.
Are there more of these workarounds? I find this concerning. There are a lot of subtle rules that apply here in the DOM spec. For example, there is the rule about namespaces, there are rules about parser context (which you don't take into account here). What about the namespace of the created tag? What about the uppercase/lowercase wrt the namespace rules of this tag?
TLDR: this looks dangerous. I guess at one point I should make my slides / content public about parser differentials and how they almost always lead to XSS or other types of injection.

Reply:
Yes let's close - we can't hope to emulate the exact behaviour of the previous parser in all edge cases, we will work around it in the Drupal test suite instead.